stac_fastapi/pgstac/stac_fastapi/pgstac/core.py

```diff
@@ -161,13 +159,16 @@ async def item_collection(
         Returns:
             An ItemCollection.
         """
+        # If collection does not exist, NotFoundError wil be raised
+        await self.get_collection(id, **kwargs)
```

Comment:
I'm curious what convinced you to safeguard against non-existent collections here. Maybe you saw some performance hits in that case which can be avoided?

Reply:
This addition is to maintain the existing logic that a request to this endpoint would result in an error if a collection doesn't exist. Previously it was the `item_collection = await self._search_base(req, **kwargs)` that would result in a 404 since no items would be found, but since that now returns an empty collection and 200 if no items are found, I wanted to make sure an error was raised. 

I was most unsure about this change, especially since The Planetary Computer STAC actually doesn't raise an error in this instance when a collection doesn't exist. I think logically it should though. What do you think?

Reply:
Good catch - the PC should def return a 404 instead of an empty collection in this case, so this change LGTM